site/cds_rdm/ldap/user_importer.py

```diff
@@ -0,0 +1,84 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2023 CERN.
+#
+# CDS-RDM is free software; you can redistribute it and/or modify it under
+# the terms of the MIT License; see LICENSE file for more details.
+
+"""CDS-RDM invenio user importer API."""
+
+from flask import current_app
+from invenio_accounts.models import User
+from invenio_db import db
+from invenio_oauthclient.models import RemoteAccount, UserIdentity
+from invenio_userprofiles.models import UserProfile
+
+
+class LdapUserImporter:
+    """Import ldap users to Invenio RDM records.
+
+    Expected input format for ldap users:
+        [
+            {'displayName': [b'Joe Foe'],
+             'department': [b'IT/CDA'],
+             'uidNumber': [b'100000'],
+             'mail': [b'joe.foe@cern.ch'],
+             'cernAccountType': [b'Primary'],
+             'employeeID': [b'101010']
+            },...
+        ]
+    """
+
+    def __init__(self):
+        """Constructor."""
+        self.client_id = current_app.config["CERN_APP_CREDENTIALS"]["consumer_key"]
+
+    def create_invenio_user(self, ldap_user):
+        """Commit new user in db."""
+        email = ldap_user["user_email"]
+        username = ldap_user["keycloak_id"]
+        user = User(email=email, username=username, active=True)
+        db.session.add(user)
+        db.session.commit()
+        return user
+
+    def create_invenio_user_identity(self, user_id, ldap_user):
+        """Return new user identity entry."""
+        uid_number = ldap_user["user_identity_id"]
+        return UserIdentity(
+            id=uid_number,
+            method=current_app.config.get("OAUTH_REMOTE_APP_NAME"),
+            id_user=user_id,
+        )
+
+    def create_invenio_user_profile(self, user, ldap_user):
+        """Return new user profile."""
+        user_profile = UserProfile(user=user)
+        user_profile.full_name = ldap_user["user_profile_full_name"]
+        return user_profile
+
+    def create_invenio_remote_account(self, user_id, ldap_user):
+        """Return new user entry."""
+        keycloak_id = ldap_user["keycloak_id"]
+        department = ldap_user["remote_account_department"]
```

Comment:
TBC if we should keep the deparment information or not

Reply:
we should since we have it